guides/06_other_tutorials/setting_up_a_gradio_demo_for_maximum_performance.md

```diff
@@ -0,0 +1,112 @@
+
+## Enabling Gradio's Queueing System
+
+By default, a Gradio demo does not use queueing and instead sends prediction requests via a POST request to the server where your Gradio server and Python code are running. However, regular POST requests have two big limitations:
+
+(1) They time out -- most browsers raise a timeout error
+if they do not get a response to a POST request after a short period of time (e.g. 1 min).
+This can be a problem if your inference function takes longer than 1 minute to run or
+if many people are trying out your demo at the same time, resulting in increased latency.
+
+(2) They do not allow bi-directional communication between the Gradio demo and the Gradio server. This means, for example, that you cannot get a real-time ETA of how long your prediction will take to complete.
+
+To address these limitations, any Gradio app can be converted to use **websockets** instead, simply by adding `.queue()` before launching an Interface or a Blocks. Here's an example:
+
+```py
+app = gr.Interface(lambda x:x, "image", "image")
+app.queue()  # <-- Sets up a queue with default parameters
+app.launch()
+```
+
+In the demo `app` above, predictions will now be sent over a websocket instead.
+Unlike POST requests, websockets do not timeout and they allow bidirectional traffic. On the Gradio server, a **queue** is set up, which adds each request that comes to a list. When a worker is free, the first available request is passed into the worker for inference. When the inference is complete, the queue sends the prediction back through the websocket to the particular Gradio user who called that prediction. 
+
+Note: If you host your Gradio app on [Hugging Face Spaces](https://hf.space), the queue is already **enabled by default**. You can still call the `.queue()` method manually in order to configure the queue parameters described below.
+
+## Queuing Parameters
+
+There are several parameters that can be used to configure the queue and help reduce latency. Let's go through them one-by-one.
+
+### The `concurrency_count` parameter
+
+The first parameter we will explore is the `concurrency_count` parameter of `queue()`. This parameter is used to set the number of worker threads in the Gradio server that will be processing your requests in parallel. By default, this parameter is set to `1` but increasing this can **linearly multiply the capacity of your server to handle requests**.
+
+So why not set this parameter much higher? Keep in mind that since requests are processed in parallel, each request will consume memory to store the data and weights for processing. This means that you might get out-of-memory errors if you increase the the `concurrency_count` too high.
+
+**Recommendation**: Increase the `concurrency_count` parameter as high as you can until you hit memory limits on your machine. You can [read about Hugging Face Spaces machine specs here](https://huggingface.co/docs/hub/spaces-overview). 
```

Comment:
Hello @abidlabs, a beautiful guide as always ❤️ 

IMO this recommendation is not a good one, because increasing concurrency does not directly translate into performance due to various reasons as costs associated with context switching and GIL limitations. So I would suggest to change it to smt like, 

'Increase concurrency as long as you see improvement in the throughput.'

Reply:
Ah I see, thanks for the suggestion @FarukOzderim! I'll update the Guide to reflect that